### PR TITLE
reorganize settings (again)

### PR DESCRIPTION
**Pull request by smichel17**
To do(ne):

- [x] Rearrange 
- [x] Changed some phrasing
- [x] Move `show hints` to the addTask screen
  - [x] Make hints disappear immediately upon unchecking the box.
- [x] Move new task capitalization to the AddTask screen.
  - [x] Add functionality in new location
  - [ ] Make setting take effect immediately instead of requiring activity restart  
@mpcjanssen I can't figure this one out
- [x] Remove pref for setting calendar reminders for threshold dates
  - [x] Remove threshold date reminder functionality
    - [ ] @vojtechkral Could you make sure my changes did the trick?

---
### Reasoning:

Having used the current settings order for a few months, there are a few items that I still consistently look in the wrong place for, despite having been the one who decided where they should go in the first place (looking at you `todo.txt terms`, `show hints`, `show datebar`, and `open landscape drawers`).

There's also some weirdness regarding hierarchy (many of the options in configuration really are not worth spending time on until you've spent a good chunk of time actually using simpletask, but they're showcased just as prominently as changing the font size or showing the datebar. Same for calendar reminders: they're useful, but they don't belong on the same level as **all** appearance-related settings.

### CODE REVIEW

**Comment by mpcjanssen** [comment]
@smichel17 I wouldn't want to remove the threshold reminder setting until we have a Lua based replacement. What is your gripe with it?

**Comment by smichel17** [comment]
Tldr a calendar is the wrong tool for the job, so when I saw an opportunity to start moving simpletask away from the business of managing a calendar, I took it. 

Some context in #466 but I ended up writing everything out here anyway :P

---
I want to make a distinction between reminders and appointments. Reminders, as their name suggests, remind you that something is on your list -- making sure you don't forget about something's **existence.** Appointments are for scheduling -- deciding **when** you are going to do something. They might be with others (eg, meet with Joe from 3-4 on Wednesday) or with yourself (eg, blocking out a couple hours to code uninterrupted). *You might have reminders about your appointments so you don't forget about them, as in the calendar, but that doesn't make appointments and reminders the same thing.*

Reminders are an important functionality of Simpletask -- you don't want to forget that you have important stuff in your list. However, we do not want to start managing appointments; that's what I, at least, mean when I say "Simpletask isn't a calendar."

Digital calendars, designed as they are to manage appointments, don't have a concept of a reminder that isn't attached to an appointment. So in order to make reminders, we are currently creating appointments, then adding reminders for those appointments, and pretending that's actually a reminder about the task.That's horribly roundabout.

This approach also sacrifices valuable functionality, the two biggest things being:

- tapping the notification to open simpletask with the relevant tasks already shown
- persistent notifications so I can't "accidentally" swipe away when something is due and needs my attention now

---

Okay, I still haven't addressed why remove threshold reminders. (One sec, switching to laptop from mobile)
